book/src/ephemeral-local-volumes.md

```diff
@@ -2,15 +2,44 @@
 
 ## Status
 
-Status | Min K8s Version | Max K8s Version 
+### CSI Ephemeral Inline Volumes
+
+Status | Min K8s Version | Max K8s Version
 --|--|--
 Alpha | 1.15 | 1.15
-Beta | 1.16 | -
+Beta | >=1.16 | -
+
+### Generic Ephemeral Inline Volumes
+
+Status | Min K8s Version | Max K8s Version
+--|--|--
+Alpha | 1.19 | -
 
 ## Overview
-Traditionally, volumes that are backed by CSI drivers can only be used with a `PersistentVolume` and `PersistentVolumeClaim` object combination. This feature supports ephemeral storage use cases and allows CSI volumes to be specified directly in the pod specification.  At runtime, nested inline volumes follow the ephemeral lifecycle of their associated pods where the driver handles all phases of volume operations as pods are created and destroyed.
+Traditionally, volumes that are backed by CSI drivers can only be used with a `PersistentVolume` and `PersistentVolumeClaim` object combination. The "CSI Ephemeral Inline volumes" feature allows CSI volumes to be specified directly in the pod specification for ephemeral use cases.  At runtime, nested inline volumes follow the ephemeral lifecycle of their associated pods where the driver handles all phases of volume operations as pods are created and destroyed.
+
+CSI drivers have to be modified to support "*CSI* Ephemeral Inline
+volumes", which is explained in more detail on this page below.  CSI
+ephemeral inline volumes are implemented locally on each node by
+kubelet talking to the CSI driver. This makes them suitable for simple
+volumes that can be created on a node.
+
+Their downsides are that none of the normal volume provisioning
+features like restoring a snapshot are supported and resource
+consumption is not considered before scheduling or at runtime of a
+pod.
+
+> See the [design document](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190122-csi-inline-volumes.md) for further design information about CSI ephemeral inline volumes.
 
-> See the [design document](https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190122-csi-inline-volumes.md) for futher information.
+Alternatively, ephemeral inline volumes can also be created with
+unmodified CSI drivers that support dynamic provisioning via
+"*generic* ephemeral inline volumes".
+
+> See the [Kubernetes
+> documentation](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volumes)
+> and the [design
+> document](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1698-generic-ephemeral-volumes)
+> for more information about generic ephemeral inline volumes.
 
 
 ### Example of inline CSI pod spec
```

Comment:
Do we need an example of generic pod spec?

Reply:
I didn't add it because I wanted to avoid further duplication of content (the Kubernetes user documentation has such an example), but I suppose for the sake of consistency on this page we have to add it. May also help understand the difference, so yes, let's add it.